src/core/tokenize.ts

```diff
@@ -240,12 +240,12 @@ function toGrammarToken (pattern: GrammarToken | RegExpLike): GrammarToken {
 }
 
 function resolve (
-	components: Registry,
+	languageRegistry: LanguageRegistry,
 	reference: Grammar | string | null | undefined
 ): Grammar | undefined {
 	if (reference) {
 		if (typeof reference === 'string') {
-			return components.getLanguage(reference);
+			return languageRegistry.getLanguage(reference)?.grammar;
```

Comment:
I have doubts about this change. I saw you left a placeholder for a `resolve()` method in the `Language` class. Is it something that should be used here instead?